File: llm_generated_code/ChatGPT/CPP/Level2/Q23.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    int t;
    cin >> t;
    
    while (t--) {
        string s;
        cin >> s;
        
        int count = 1;
        char current = s[0];
        int items = 0;
        
        for (int i = 0; i < s.size(); i++) {
            if (s[i] == current) {
                items++;
            } else {
                current = s[i];
                items = 1;
            }
            
            if (items == 5) {
                count++;
                items = 0;
            }
        }
        
        cout << count << endl;
    }
    
    return 0;
}

File: llm_generated_code/ChatGPT/Python/Level2/Q23.py
def main():
    n, m = map(int, input().split())
    items = input().strip()
    count = 0
    i = 0
    while i < n:
        count += 1
        item_count = 0
        current_item = items[i]
        while i < n and item_count < m and items[i] == current_item:
            item_count += 1
            i += 1
    print(count)

if __name__ == "__main__":
    main()
